cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/index.js

```diff
@@ -0,0 +1,323 @@
+  }
+
+  getInheritedPreferences() {
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    myPreferenceApi.getSystemPreference()
+    .subscribe(
+     (systemRes) => {
+       let systemPrefArray = this.getKeyValPair(systemRes, 'system');
+       myPreferenceApi.getNamespacePreference({namespace})
+      .subscribe(
+       (namespaceRes) => {
+         let namespacePrefArray = this.getKeyValPair(namespaceRes, 'namespace');
+         let prefArray = systemPrefArray.concat(namespacePrefArray);
+         this.setState({
+           inheritedPreferences: prefArray
+         });
+       },
+       (namespaceError) => {
+         this.setState({
+           error: isObject(namespaceError) ? namespaceError.response : namespaceError
+         });
+       }
+      );
+     },
+     (systemError) => {
+       this.setState({
+         error: isObject(systemError) ? systemError.response : systemError
+       });
+     }
+    );
+  }
+
+  saveAppPreferences(event) {
+    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    myPreferenceApi.setAppPreference({namespace, appId: this.appId}, this.getKeyValObject())
+    .subscribe(
+      (error) => {
+        this.setState({
+          error: isObject(error) ? error.response : error
+        });
+      }
+    );
+    if (event) {
+      event.stopPropagation();
+      event.nativeEvent.stopImmediatePropagation();
+    }
+  }
+
+  getKeyValPair(prefObj, origin = '') {
+    let prefArray = [];
+    for (let key in prefObj) {
+      if (prefObj.hasOwnProperty(key)) {
+        prefArray.push({key: key, value: prefObj[key], uniqueId: shortid.generate(), origin: upperFirst(origin)});
+      }
+    }
+    return prefArray;
+  }
+
+  getKeyValObject() {
+    let keyValArr = this.state.keyValues.pairs;
+    let keyValObj = {};
+    keyValArr.forEach((pair) => {
+      keyValObj[pair.key] = pair.value;
+    });
+    return keyValObj;
+  }
+
+  onKeyValueChange(keyValues) {
+    this.setState({keyValues});
+  }
+
+  componentWillMount() {
+    this.getAppPreferences();
+    this.getInheritedPreferences();
+  }
+
+  renderSpecifyPreferences() {
+    return (
+      <div>
+        <div className = 'specify-preferences-description'>
+          <h4>Specify Preferences</h4>
+          <p>Specify new or override existing system or namespace preferences. These preferences will be accessible in all programs within this application.</p>
+        </div>
+        <div className = 'specify-preferences-list'>
+          <div className = 'specify-preferences-labels'>
+            <span className = 'key-label'>KEY</span>
+            <span className = 'value-label'>VALUE</span>
+          </div>
+          <KeyValuePairs
+            keyValues = {this.state.keyValues}
+            onKeyValueChange = {this.onKeyValueChange}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  renderInheritedPreferences() {
+    return (
+      <div>
+        <div className = 'inherited-preferences-label'>
```

Comment:
Space after `className` doesn't seem to be consistent across everywhere. The existing pattern is to not have any space.